getMultiStarts.m

```diff
@@ -222,40 +234,51 @@
         error_count = 0;
         
         % Test evaluation of objective function at starting point
+        % Only for multi-start local, since other optimizers use a 
+        % different initialization
         if (strcmp(options.localOptimizer, 'fmincon'))
             if (strcmp(options.localOptimizerOptions.Hessian, 'on'))
-                [J_0,~,~] = negLogPostWErrorCount(parameters.MS.par0(:,i)); % objectiveWrapWErrorCount(parameters.MS.par0(:,i),objective_function,options.obj_type,options.objOutNumber);
+                % We have to check Hessian AND gradient computation, since
+                % simulation can give different results if we're close to
+                % numerical failure
+                [J_0,~,~] = negLogPostWErrorCount(parameters.MS.par0(:,iMS));
```

Comment:
It is a bugfix for a super specific problem that cannot be expected to generalize but generally increases additional computation time for uses of PESTO? That sounds like pretty bad deal. (Also the 200 additional gradient evaluations might just net you close to 4 additional multistarts in your setting)

If we continually do this, PESTO will become and patchwork abomination of small fixes that individually only marginally increase computation time but make code hard to understand and increasingly slow.